### ISSUE TITLE
How to pass derivation as source to makeFlakeOutputs?

### ISSUE THREAD

**Issue by zmrocze** [nix-community/dream2nix#518]
Hey, 

## TLDR

Ideally I'd like to pass derivation as `source` to `makeFlakeOutputs`, like here:

```nix
# node js project
frontend = dream2nix.lib.makeFlakeOutputs {
  systems = [ system ];
  config.projectRoot = ./frontend;
  source = frontendFullSrc;  # a derivation producing the source 
  projects = ./frontend/project.toml;
};
```

but the source argument is expected to be a path, see [implementation](https://github.com/nix-community/dream2nix/blob/fbfb09d2ab5ff761d822dd40b4a1def81651d096/src/modules/dlib/implementation.nix#L51).

I guess I'd need to use `packageOverrides` functionality instead to achieve something similar, do I? How could I use it?

## Problem description

I have a javascript project. I generate one javascript module and include it into the source, this module is untracked by git.
I'll demonstrate with the current definition of `frontendFullSrc` from the above snippet:

```nix
# Derivation producing frontend's src together with src/Generated.js
frontendFullSrc = pkgs.symlinkJoin "frontend" [
  ./frontend            # source subdirectory, 
  (pkgs.runCommandLocal "src/Generated.js" {} ''
    mkdir src
    mv ${generated-module} src
    cp -r src $out
  '')
];
```

so my question is:

### How do I inject some code into source with the current interface?

I guess `packageOverrides` would be used for that. But I had problem previously using `packageOverrides` to append `buildInputs` - it seemed like these options had no effect. So I would very much appreciate someone giving me a hint on how to do that (doesn't have to be crystal clear, a source link would do afaic). 

### Feature request: Could a derivation be used for the `source` `makeFlakeOutputs` argument, optionally to strings/paths?

so that the first snippet works

Thanks!

**Comment by zmrocze** [comment]
# SOLVED

It seems the answer is simply: "You can pass derivation as ``source`". I glanced over the definition of `toString`, the word derivation is not there, so that's the source of my confusion.

I'll close once I confirm it's indeed the case with some e2e success
